docs/20-guides/101-datocms/index.md

```diff
@@ -0,0 +1,272 @@
+
+::: tip
+Useful resources:
+- [Repeating elements](formulas/repeat-formula)
+- [Understanding when to use components](guides/component-creation)
+:::
+
+## Creating the Article Page
+
+The article page requires two primary configurations: fetching the complete blog post data and setting up the repeating block component.
+
+On the article page in Nordcraft, create a page parameter named **slug**.
+
+The blog post’s slug, which serves as a unique identifier for each article, will be used to format the query.
+
+The final query should be structured as follows:
+
+```
+{
+  allPosts(filter: {slug: {eq: "nextgen-image-formats-webp-and-avif"}}) {
+    title
+    seoTags {
+      image {
+        url
+        alt
+      }
+    }
+    author {
+      name
+      picture {
+        url
+      }
+      bio
+    }
+    _publishedAt
+    content {
+      blocks {
+        id
+        image {
+          id
+          alt
+          url
+        }
+      }
+      links {
+        id
+        slug
+        title
+      }
+      value
+    }
+  }
+}
+```
+
+This query retrieves the article's title, main image, author profile, and content.
+
+Now, examine the **dato-block** component, which is a recursive component iterating over the article's content.
+
+### Recursive Content Component
+
+Recursion is when a process calls itself to solve smaller or "nested" versions of the same problem until an end or **base case** is reached. It is useful for things like tree traversals or, in this case, going through the content array and its nested arrays to display it as a blog article.
+
+Recursive components cans sometimes be challenging to conceptualize so here is a breakdown of its structure and functionality.
+
+To understand its setup, first review the content data:
+
+![Nested Children|16/9](nested-children.webp)
+
+The initial “children” array is the target for repeating the **dato-block** component. Note that its first index item also contains its own “children” array.
+
+This structure is typical for headless CMS systems, with minor variations across platforms. The purpose of this nesting is to separate inline elements such as bold text or links as seen in the paragraph example.
+
+::: info
+Some CMS platforms return regular HTML, but JSON is seen more often. JSON is much easier to work with on the front end, as it allows you to fully customise the HTML structure of your application.
+:::
+
+In the example above, the content includes regular text, bold text, another segment of regular text, and italics. These inline elements form the complete paragraph, and it is these nested children elements that the component iterates over (2). This is the "nested" problem that we are solving, a chain of arrays (elements) with their own child arrays (inline elements).
```

Comment:
I guess these numbers refer to areas on the images? Sorry I haven't looked at this PR on the front end preview yet :D